pkg/handler/queue_test.go

```diff
@@ -0,0 +1,109 @@
+/*
+	Copyright 2019 whiteblock Inc.
+	This file is a part of the genesis.
+
+	Genesis is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Genesis is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package handler
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/streadway/amqp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/whiteblock/genesis/mocks"
+	"github.com/whiteblock/genesis/pkg/command"
+	"github.com/whiteblock/genesis/pkg/entity"
+	"github.com/whiteblock/genesis/pkg/usecase"
+)
+
+func TestDeliveryHandler_ProcessMessage(t *testing.T) {
```

Comment:
Can you test that it calls Run on the usecase and provides the expected command as input?

Reply:
Mock the usecase, but don't mock the handler in the test. Also, we should only go one level deep, no need to mock the services. The tests for the usecase will cover that.